make/modules/java.desktop/lib/Awt2dLibraries.gmk

```diff
@@ -832,47 +806,47 @@ ifeq ($(call isTargetOs, macosx), true)
       libosxapp \
       #
 
-  LIBAWT_LWAWT_CFLAGS := $(X_CFLAGS) $(X_LIBS)
-
-  LIBAWT_LWAWT_EXFILES := fontpath.c awt_Font.c X11Color.c
+  LIBAWT_LWAWT_EXCLUDE_FILES := fontpath.c awt_Font.c X11Color.c
   LIBAWT_LWAWT_EXCLUDES := $(TOPDIR)/src/$(MODULE)/unix/native/common/awt/medialib
 
   $(eval $(call SetupJdkLibrary, BUILD_LIBAWT_LWAWT, \
       NAME := awt_lwawt, \
       EXTRA_SRC := $(LIBAWT_LWAWT_EXTRA_SRC), \
-      INCLUDE_FILES := $(LIBAWT_LWAWT_FILES), \
-      EXCLUDE_FILES := $(LIBAWT_LWAWT_EXFILES), \
+      EXCLUDE_FILES := $(LIBAWT_LWAWT_EXCLUDE_FILES), \
       EXCLUDES := $(LIBAWT_LWAWT_EXCLUDES), \
       OPTIMIZATION := LOW, \
-      CFLAGS := $(LIBAWT_LWAWT_CFLAGS), \
       EXTRA_HEADER_DIRS := $(LIBAWT_LWAWT_EXTRA_HEADER_DIRS), \
-      DISABLED_WARNINGS_clang := incomplete-implementation deprecated-declarations \
-          objc-method-access incompatible-pointer-types extra-tokens sign-compare undef, \
+      DISABLED_WARNINGS_clang := incomplete-implementation \
+          deprecated-declarations objc-method-access \
+          incompatible-pointer-types extra-tokens sign-compare undef, \
       DISABLED_WARNINGS_clang_AWTWindow.m := bitwise-op-parentheses, \
       DISABLED_WARNINGS_clang_CFileDialog.m := semicolon-before-method-body, \
       DISABLED_WARNINGS_clang_CGGlyphImages.m := pointer-arith, \
       DISABLED_WARNINGS_clang_CGLLayer.m := semicolon-before-method-body, \
-      DISABLED_WARNINGS_clang_ImageSurfaceData.m := enum-conversion parentheses-equality, \
+      DISABLED_WARNINGS_clang_ImageSurfaceData.m := enum-conversion \
+          parentheses-equality, \
       DISABLED_WARNINGS_clang_MTLBlitLoops.m := pointer-arith, \
-      DISABLED_WARNINGS_clang_MTLPipelineStatesStorage.m := semicolon-before-method-body, \
+      DISABLED_WARNINGS_clang_MTLPipelineStatesStorage.m := \
+          semicolon-before-method-body, \
       DISABLED_WARNINGS_clang_MTLRenderer.m := gnu-folding-constant, \
       DISABLED_WARNINGS_clang_MTLVertexCache.m := pointer-arith, \
       DISABLED_WARNINGS_clang_OGLBufImgOps.c := format-nonliteral, \
       DISABLED_WARNINGS_clang_OGLPaints.c := format-nonliteral, \
       DISABLED_WARNINGS_clang_PrinterView.m := enum-conversion, \
       LDFLAGS := -L$(INSTALL_LIBRARIES_HERE), \
-      LIBS := -lawt -lmlib_image -losxapp -ljvm $(LIBM) \
+      JDK_LIBS_macosx := -lawt -ljava -ljvm -lmlib_image -losxapp, \
+      LIBS_macosx := $(LIBM) \
           -framework Accelerate \
           -framework ApplicationServices \
           -framework AudioToolbox \
           -framework Carbon \
           -framework Cocoa \
-          -framework Metal \
-          -framework Security \
           -framework ExceptionHandling \
           -framework JavaRuntimeSupport \
+          -framework Metal \
           -framework OpenGL \
-          -framework QuartzCore -ljava, \
+          -framework QuartzCore \
```

Comment:
The re-ordering here happens to be small, and it looks like some one already might have been doing somethings lexically, but this again points to the need for running tests as well as building.

Reply:
Here, as in many places, there have been attempts at keeping the lists sorted, but unfortunately this has not been kept 100%. :-(

The extraction of -ljava is essential for the upcoming JDK_LIB rewrite.

Once again, if you have any suggestions to what tests you think I should run, let me know!